examples/thermostat/thermostat-common/include/atomic-write-manager.h

```diff
@@ -0,0 +1,186 @@
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/clusters/thermostat-server/atomic-write.h>
+#include <app/clusters/thermostat-server/thermostat-delegate.h>
+
+#include "thermostat-delegate-impl.h"
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace Thermostat {
+
+enum AtomicWriteState
+{
+    kAtomicWriteState_Closed = 0,
+    kAtomicWriteState_Open,
+};
+
+class ThermostatAtomicWriteManager : public AtomicWriteManager
+{
+public:
+    static inline ThermostatAtomicWriteManager & GetInstance() { return sInstance; }
+
+    /**
+     * @brief Set the Delegate object
+     *
+     * @param delegate
+     */
+    void SetDelegate(AtomicWriteDelegate * delegate) override { mDelegate = delegate; };
+
+    /**
+     * @brief Check if there is an open atomic write on an endpoint, optionally associated with a specific attribute
+     *
+     * @param attributeId Optional attribute filter
+     * @param endpoint The endpoint to check atomic write state
+     * @return true if there is an open atomic write
+     * @return false if there is not at open atomic write
+     */
+    bool InWrite(const std::optional<AttributeId> attributeId, EndpointId endpoint) override;
+
+    /**
+     * @brief Check if there is an open atomic write on an endpoint, associated with a given SubjectDescriptor, and optionally
+     * associated with a specific attribute
+     *
+     * @param attributeId Optional attribute filter
+     * @param subjectDescriptor The subject descriptor to check against
+     * @param endpoint The endpoint to check atomic write state
+     * @return true if there is an open atomic write
+     * @return false if there is not an open atomic write
+     */
+    bool InWrite(const std::optional<AttributeId> attributeId, const Access::SubjectDescriptor & subjectDescriptor,
+                 EndpointId endpoint) override;
+
+    /**
+     * @brief Check if there is an open atomic write on an endpoint, associated with the source node for a given command invocation,
+     * and optionally associated with a specific attribute
+     *
+     * @param attributeId
+     * @param commandObj
+     * @param endpoint
+     * @return true
+     * @return false
+     */
+    bool InWrite(const std::optional<AttributeId> attributeId, CommandHandler * commandObj, EndpointId endpoint) override;
+
+    /**
+     * @brief Check if there is an open atomic write on an endpoint, associated with the source node for a given command invocation,
+     * and associated with the set of attribute IDs
+     *
+     * @param attributeIds The set of attribute IDs to check against
+     * @param commandObj The command being invoked
+     * @param endpoint The endpoint for the atomic write
+     * @return true if there is an open atomic write
+     * @return false if there is not an open atomic write
+     */
+    bool InWrite(const DataModel::DecodableList<AttributeId>::Iterator attributeIds, CommandHandler * commandObj,
```

Comment:
could we have one api with an optional list of attributeids and then the case where we need one attribute id could be a list of length 1?

Reply:
same applies to other sets of APIs